### PR TITLE
Migrate WebPubSub to azure-json

### PR DESCRIPTION
**Pull request by alzimmermsft**
# Description

Migrates `azure-messaging-webpubsub` and `azure-messaging-webpubsub-client` to `azure-json`.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/webpubsub/azure-messaging-webpubsub-client/src/main/java/com/azure/messaging/webpubsub/client/WebPubSubAsyncClient.java [hunk 1]
```diff
@@ -1086,27 +1035,26 @@ WebSocketSession getWebsocketSession() {
...
-        String errorMessage, Throwable cause, boolean isTransient, Long ackId) {
+    private RuntimeException logSendMessageFailedException(String errorMessage, Throwable cause, boolean isTransient,
+        Long ackId) {
 
         return logSendMessageFailedException(errorMessage, cause, isTransient, ackId, null);
     }
 
-    private RuntimeException logSendMessageFailedException(
-        String errorMessage, Throwable cause, boolean isTransient, Long ackId, AckResponseError error) {
+    private RuntimeException logSendMessageFailedException(String errorMessage, Throwable cause, boolean isTransient,
```
Comment (haolingdong-msft) [inline sdk/webpubsub/azure-messaging-webpubsub-client/src/main/java/com/azure/messaging/webpubsub/client/WebPubSubAsyncClient.java:1056]: There are many changes in this file, are these all formatting changes, or is there code change?

File: sdk/webpubsub/azure-messaging-webpubsub-client/src/test/java/com/azure/messaging/webpubsub/client/ClientTests.java [hunk 2]
```diff
@@ -42,18 +42,15 @@ public void testClientState() {
```
Comment (haolingdong-msft) [inline sdk/webpubsub/azure-messaging-webpubsub-client/src/test/java/com/azure/messaging/webpubsub/client/ClientTests.java:42]: Have we run the live tests? Guessing we need to run the live tests and re-record to make sure the seriaization is expected. Add @weidongxu-microsoft @srnagar  for suggestion.

File: sdk/webpubsub/azure-messaging-webpubsub-client/src/test/java/com/azure/messaging/webpubsub/client/GroupMessageTests.java [hunk 3]
```diff
@@ -190,8 +198,33 @@ public void testSendMessagePerformance() throws InterruptedException {
         }
     }
 
-    private static class JsonModel {
+    private static class JsonModel implements JsonSerializable<JsonModel> {
```
Comment (weidongxu-microsoft) [inline sdk/webpubsub/azure-messaging-webpubsub-client/src/test/java/com/azure/messaging/webpubsub/client/GroupMessageTests.java:201]:
```suggestion
    public static class JsonModel implements JsonSerializable<JsonModel> {
```

`java.lang.IllegalStateException: java.lang.IllegalAccessException` is private

File: sdk/webpubsub/azure-messaging-webpubsub-client/src/test/java/com/azure/messaging/webpubsub/client/ClientTests.java (hunk 2 above)
Comment (weidongxu-microsoft) [reply sdk/webpubsub/azure-messaging-webpubsub-client/src/test/java/com/azure/messaging/webpubsub/client/ClientTests.java:42]: I've tested azure-messaging-webpubsub-client, the only problem is https://github.com/Azure/azure-sdk-for-java/pull/40718/files#r1646838102

After this change all passes.

Comment (weidongxu-microsoft) [reply sdk/webpubsub/azure-messaging-webpubsub-client/src/test/java/com/azure/messaging/webpubsub/client/ClientTests.java:42]: I will try make a PR to switch azure-messaging-webpubsub-client away from CONNECTION_STRING.

**Review by weidongxu-microsoft — APPROVED** [review]

File: sdk/webpubsub/azure-messaging-webpubsub-client/src/main/java/com/azure/messaging/webpubsub/client/WebPubSubAsyncClient.java (hunk 1 above)
Comment (alzimmermsft) [reply sdk/webpubsub/azure-messaging-webpubsub-client/src/main/java/com/azure/messaging/webpubsub/client/WebPubSubAsyncClient.java:1056]: This was just the Eclipse formatting rules getting applied to this file. I can revert them if wanted

**Pull request merged**